packages/context/src/provider.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+
+/**
+ * Interface definition for a MetaValueProvider
+ *
+ * MetaValueProvider provisions injecting values via the constructor and
+ * returns a computed value (Object or Function) when the value() function is called
+ *
+ * @example:
+ * ```ts
+ * export class MyProvider implements MetaValueProvider {
+ *   constructor(@inject('value') param){}
+ *
+ *   static key = 'MY_KEY';
+ *
+ *   value() {
+ *     return computed_value;
+ *   }
+ * }
+ *
+ * ```
+ */
+export interface MetaValueProvider {
+  key: string;
+  value(): Object | Function;
```

Comment:
I think the signature should be: `value(): BoundValue | Promise<BoundValue>;` so that the value can be produced synchronously or asynchronously.